### ISSUE TITLE
👀 How to handling error

### ISSUE THREAD

**Issue by himprover** [himprover/minfolink-client#39]
## What is error handling?
'Error Handling' in this document refers to the way of dealing with errors that occur during asynchronous communication with the server.

## What kind of situations can occur when asynchronous communication
In this project, three types of erros are assumed.

1. An error occurred on the client before making a request to the server.
2. An error occurred on the server after making a request and receiving a response
3. An unexpected error occurred.

## How to handle them?
To handle these erros, they must be distinguished first. The errors are categorized into two types: **whether they are caused by the request or not.**

In this project, asynchronous requests are made using `axios`. Therefore, it can be distinguished whether the error occurred is an `axios` error or not, by checking its types.

To summarize, in case of errors occurred during the request process, we distinguish between axios errors and others, and handle them as either 4xx or 5xx errors. Error boundary handles errors by referring to the error class defined internally.
